src/extensions/registries/page-registry.ts

```diff
@@ -20,12 +23,26 @@ export interface PageComponents {
   Page: React.ComponentType<any>;
 }
 
+const routePrefix = "/extension/:name"
+
+export function getPageUrl(ext: LensExtension, baseUrl = "") {
+  const validUrlName = ext.name.replace("@", "").replace("/", "-");
+  return compile(routePrefix)({ name: validUrlName }) + baseUrl;
+}
+
 export class PageRegistry<T extends PageRegistration> extends BaseRegistry<T> {
-  getItems() {
-    return super.getItems().map(item => {
-      item.routePath = item.extension.getPageRoute(item.routePath)
-      return item
-    });
+
+  @action
+  add(items: T[], ext?: LensExtension) {
+    const normalizedItems = items.map((i) => {
+      i.routePath = getPageUrl(ext, i.routePath)
```

Comment:
`subPages` not handled (see my PR)

Reply:
`subPages` & `subMenus` were totally broken, those are now removed. Let's put them back once we actually have better idea how those should be implemented.